DQM/SiStripMonitorDigi/src/SiStripMonitorDigi.cc

```diff
@@ -606,31 +627,64 @@ void SiStripMonitorDigi::analyze(const edm::Event& iEvent, const edm::EventSetup
 
     uint16_t iDet = 0;
 
-    std::string subdet_label = ""; 
+    std::string subdet_label = "";
 
     // loop over all modules in the layer
-    for (std::vector< uint32_t >::const_iterator iterDets = layer_dets.begin() ; 
+    for (std::vector< uint32_t >::const_iterator iterDets = layer_dets.begin() ;
 	 iterDets != layer_dets.end() ; iterDets++) {
       iDet++;
-      
+
       // detid and type of ME
       uint32_t detid = (*iterDets);
-	
+
       // Get SubDet label once
       if (subdet_label.size() == 0) subdet_label = folder_organizer.getSubDetFolderAndTag(detid, tTopo).second;
 
       // DetId and corresponding set of MEs
 
       std::map<uint32_t, ModMEs >::iterator pos = DigiMEs.find(detid);
       ModMEs local_modmes = pos->second;
-	
+
       // search  digis of detid
-      int loc = getDigiSourceIndex(detid); 
-      
+      int loc = getDigiSourceIndex(detid);
+
       int ndigi_det = 0;
-      
-      if (loc > -1) {	
+
+      if (loc > -1) {
 	ndigi_det = (*(digi_detset_handles[loc]))[detid].size();
+
+  //////////////////////// FED N digis Monitoring //////////////////////
+
+//Get all FED connections associated with given detID.
+// All connections for a detid have same FED Id therefore one FEDID is associated with a given detID.
+std::vector<const FedChannelConnection*> fedConnections = SiStripDetCabling_->getConnections(detid);
+
+// Filling FED Id associated clusters map.
+
+int good_fcc_index = -999;
+for(uint x=0;x<fedConnections.size();x++){
```

Comment:
hi @Wilsker - similar issues as already fixed here - change uint to unsigned_int and no need for the at

in fact all occurrences of this can be replaced by:

for ( unsigned int x=0; x<fedConnecctions.size(); x++) {
 if ( fedConnections[x] != nullptr ) {
   good_fcc_index=x;
   break;
}
}